packages/camera/lib/src/camera.dart

```diff
@@ -0,0 +1,454 @@
+          'resolutionPreset': serializeResolutionPreset(resolutionPreset),
+        },
+      );
+      _textureId = reply['textureId'];
+      value = value.copyWith(
+        isInitialized: true,
+        previewSize: Size(
+          reply['previewWidth'].toDouble(),
+          reply['previewHeight'].toDouble(),
+        ),
+      );
+    } on PlatformException catch (e) {
+      throw CameraException(e.code, e.message);
+    }
+    _eventSubscription =
+        EventChannel('flutter.io/cameraPlugin/cameraEvents$_textureId')
+            .receiveBroadcastStream()
+            .listen(_listener);
+    _creatingCompleter.complete();
+    return _creatingCompleter.future;
+  }
+
+  /// Listen to events from the native plugins.
+  ///
+  /// A "cameraClosing" event is sent when the camera is closed automatically by the system (for example when the app go to background). The plugin will try to reopen the camera automatically but any ongoing recording will end.
+  void _listener(dynamic event) {
+    final Map<dynamic, dynamic> map = event;
+    if (_isDisposed) {
+      return;
+    }
+
+    switch (map['eventType']) {
+      case 'error':
+        value = value.copyWith(errorDescription: event['errorDescription']);
+        break;
+      case 'cameraClosing':
+        value = value.copyWith(isRecordingVideo: false);
+        break;
+    }
+  }
+
+  /// Captures an image and saves it to [path].
+  ///
+  /// A path can for example be obtained using
+  /// [path_provider](https://pub.dartlang.org/packages/path_provider).
+  ///
+  /// If a file already exists at the provided path an error will be thrown.
+  /// The file can be read as this function returns.
+  ///
+  /// Throws a [CameraException] if the capture fails.
+  Future<void> takePicture(String path) async {
+    if (!value.isInitialized || _isDisposed) {
+      throw CameraException(
+        'Uninitialized CameraController.',
+        'takePicture was called on uninitialized CameraController',
+      );
+    }
+    if (value.isTakingPicture) {
+      throw CameraException(
+        'Previous capture has not returned yet.',
+        'takePicture was called before the previous capture returned.',
+      );
+    }
+    try {
+      value = value.copyWith(isTakingPicture: true);
+      await _channel.invokeMethod(
+        'takePicture',
+        <String, dynamic>{'textureId': _textureId, 'path': path},
+      );
+      value = value.copyWith(isTakingPicture: false);
+    } on PlatformException catch (e) {
+      value = value.copyWith(isTakingPicture: false);
+      throw CameraException(e.code, e.message);
+    }
+  }
+
+  /// Start a streaming bytes from platform camera.
+  ///
+  /// The stream will always use the latest image and discard the others.
+  ///
+  /// [onAvailable] is a method that takes a [CameraImage] and
+  /// returns `void`.
+  ///
+  /// Throws a [CameraException] if byte streaming or video recording has
+  /// already started.
```

Comment:
Maybe discuss the performance-implications of this (I have not tried running this, so I don't know it is as bad as I fear)

Reply:
I included some performance notes when using it with CameraPreview. It can slow down it's performance when streaming images. However, this PR is paired with the ML Kit image byte data PR: https://github.com/flutter/plugins/pull/971. And through testing, it runs quite well.